components/sync15/src/state.rs

```diff
@@ -380,7 +459,14 @@ impl<'a> SetupStateMachine<'a> {
                 self.client.wipe_all_remote()?;
 
                 // Upload a fresh `meta/global`...
+                let (mut changes, _) = self.pgs.update_engine_state(None, self.engine_updates);
+                // Already wiped.
+                changes.remote_wipes.clear();
+
+                self.changes_needed = Some(changes);
```

Comment:
Hmm, do we need to set this here, too? IIUC, after `FreshStartRequired`, we'll go to `InitialWithConfig` -> `InitialWithInfo`, fetch `m/g` again, and update `changes_needed` there?

Reply:
I think so? We shouldn't find any changes needed the 2nd time through.